### PR TITLE
Fix spelling in comments

### PR DESCRIPTION
**Pull request by mdavidsaver**
The charactor of my speling has been impuned!  So with this change I correct in bulk most (although no all) of the spelling errors in C/C++ code comments.  This should be practically non-functional, with the exception of some error message strings.

I've gone through the diff manually and sorted out a few incorrect changes to code examples in comments.  The raw replacements are [here](https://github.com/epics-base/epics-base/files/6988555/fix.json.txt).

For the curious.  Although only touching ~400 lines, this change touches ~20% of the C/C++ code files in Base.

@anjohnson I used an en_US dictionary, so apologies in advance for any words which are now being spelled incorrectly.

### CODE REVIEW

File: modules/database/src/ioc/rsrv/camessage.c [hunk 1]
```diff
@@ -503,7 +503,7 @@ static void read_reply ( void *pArg, struct dbChannel *dbch,
     cid = ECA_NORMAL;
 
     /* If the client has requested a zero element count we interpret this as a
-     * request for all avaiable elements.  In this case we initialise the
+     * request for all available elements.  In this case we initialism the
```
Comment (anjohnson) [inline modules/database/src/ioc/rsrv/camessage.c]: I think you meant `initialize` here.

File: modules/libcom/src/cxxTemplates/resourceLib.h [hunk 2]
```diff
@@ -1000,7 +1000,7 @@ inline void chronIntIdRes<ITEM>::setId (unsigned newId)
 //
 // intId::intId
 //
-// (if this is inline SUN PRO botches the template instantiation)
+// (if this is inline SUN PRO botches the template instantiate)
```
Comment (anjohnson) [inline modules/libcom/src/cxxTemplates/resourceLib.h]: The original was correct here, could that word be missing from your dictionary?

File: modules/libcom/src/fdmgr/fdManager.cpp [hunk 3]
```diff
@@ -177,8 +177,8 @@ LIBCOM_API void fdManager::process (double delay)
         else if ( status < 0 ) {
             int errnoCpy = SOCKERRNO;
 
-            // dont depend on flags being properly set if
-            // an error is retuned from select
+            // don't depend on flags being properly set if
+            // an error is re-tuned from select
```
Comment (anjohnson) [inline modules/libcom/src/fdmgr/fdManager.cpp]: I think this should be `returned`.

File: modules/libcom/src/flex/ccl.c [hunk 4]
```diff
@@ -14,7 +14,7 @@
  * Redistribution and use in source and binary forms are permitted provided
  * that: (1) source distributions retain this entire copyright notice and
  * comment, and (2) distributions including binaries display the following
- * acknowledgement:  ``This product includes software developed by the
+ * acknowledgment:  ``This product includes software developed by the
```
Comment (anjohnson) [inline modules/libcom/src/flex/ccl.c]: This is part of the legal license text which must be retained, and I would hesitate to change it; the original spelling is the preferred British form (was UCB's lawyer a Brit?). This comment applies to all copies of this notice in the modules/libcom/src/flex directory which all have the same spelling.

File: modules/libcom/src/flex/dfa.c [hunk 5]
```diff
@@ -830,7 +830,7 @@ int snstods(int sns[], int numstates, int accset[], int nacc, int hashval, int *
 
     else if ( reject )
         {
-        /* we sort the accepting set in increasing order so the disambiguating
+        /* we sort the accepting set in increasing order so the disambiguate
```
Comment (anjohnson) [inline modules/libcom/src/flex/dfa.c]: "disambiguating rule" – a rule for removing ambiguity.
Another originally correct spelling that seems to be missing from your dictionary, the same change also needs to be undone 22 lines below in the same file.

File: modules/libcom/src/flex/tblcmp.c [hunk 6]
```diff
@@ -865,7 +865,7 @@ void stack1(int statenum, int sym, int nextstate, int deflink)
  *
  * "state" is the state array which is to be extracted from the pr'th
  * proto.  "pr" is both the number of the proto we are extracting from
- * and an index into the save area where we can find the proto's complete
+ * and an index into the save area where we can find the protos complete
```
Comment (anjohnson) [inline modules/libcom/src/flex/tblcmp.c]: Possessive, belonging to or of the proto. Original is correct.

File: modules/libcom/src/osi/compiler/default/compilerSpecific.h [hunk 7]
```diff
@@ -17,7 +17,7 @@
 #ifndef compilerSpecific_h
 #define compilerSpecific_h
 
-/* The 'inline' key work, possibily with compiler
+/* The 'inline' key work, possibly with compiler
```
Comment (anjohnson) [inline modules/libcom/src/osi/compiler/default/compilerSpecific.h]: This change is correct, but I think `key work` should be `keyword`.

File: modules/libcom/src/osi/compiler/gcc/compilerSpecific.h [hunk 8]
```diff
@@ -33,9 +33,9 @@
 #ifdef __cplusplus
 
 /*
- * in general we dont like ifdefs but they do allow us to check the
+ * in general we don't like ifdefs but they do allow us to check the
  * compiler version and make the optimistic assumption that
- * standards incompliance issues will be fixed by future compiler
+ * standards in compliance issues will be fixed by future compiler
```
Comment (anjohnson) [inline modules/libcom/src/osi/compiler/gcc/compilerSpecific.h]: **incompliance** (noun) – The condition of being incompliant; a refusal to comply.

Another dictionary hole. You could change the word to non-compliance if that makes the meaning clearer.

File: modules/libcom/src/osi/compiler/msvc/compilerSpecific.h [hunk 9]
```diff
@@ -29,9 +29,9 @@
 #ifdef __cplusplus
 
 /*
- * in general we dont like ifdefs but they do allow us to check the
+ * in general we don't like ifdefs but they do allow us to check the
  * compiler version and make the optimistic assumption that
- * standards incompliance issues will be fixed by future compiler
+ * standards in compliance issues will be fixed by future compiler
```
Comment (anjohnson) [inline modules/libcom/src/osi/compiler/msvc/compilerSpecific.h]: See above.

File: modules/libcom/src/osi/os/WIN32/epicsGetopt.c [hunk 10]
```diff
@@ -13,7 +13,7 @@
  *    notice, this list of conditions and the following disclaimer in the
  *    documentation and/or other materials provided with the distribution.
  * 3. All advertising materials mentioning features or use of this software
- *    must display the following acknowledgement:
+ *    must display the following acknowledgment:
```
Comment (anjohnson) [inline modules/libcom/src/osi/os/WIN32/epicsGetopt.c]: Another legal notice from UCB which shouldn't be fixed.

File: modules/libcom/src/osi/os/vxWorks/task_params.h [hunk 11]
```diff
@@ -55,9 +55,9 @@
 /* Task priorities */
 #define SCANONCE_PRI    129      /* scan one time */
 /*DO NOT RUN ANY RECORD PROCESSING TASKS AT HIGHER PRIORITY THAN _netTask=50*/
-#define CALLBACK_PRI_LOW 140      /* callback task - generall callback task */
-#define CALLBACK_PRI_MEDIUM 135     /* callback task - generall callback task */
-#define CALLBACK_PRI_HIGH 128      /* callback task - generall callback task */
+#define CALLBACK_PRI_LOW 140      /* callback task - generally callback task */
+#define CALLBACK_PRI_MEDIUM 135     /* callback task - generally callback task */
+#define CALLBACK_PRI_HIGH 128      /* callback task - generally callback task */
```
Comment (anjohnson) [inline modules/libcom/src/osi/os/vxWorks/task_params.h]: These three should all say `general`.

File: modules/libcom/src/osi/osiSock.h [hunk 12]
```diff
@@ -203,7 +203,7 @@ LIBCOM_API void epicsStdCall osiSockDiscoverBroadcastAddresses
  * repeater would not always allow the loopback address
  * as a local client address so current clients alternate
  * between the address of the first non-loopback interface
- * found and the loopback addresss when subscribing with
+ * found and the loopback addresses when subscribing with
```
Comment (anjohnson) [inline modules/libcom/src/osi/osiSock.h]: Surely there's only one loopback address, so this should be singular.

**Review by anjohnson — CHANGES_REQUESTED** [review]
Wow! Have you been having fun with this? I'm pretty sure Jeff still beat you in the total number of mis-spellings though.

I found a few small issues and dictionary holes, see the comments inline. I'm only complaining about the loss of British spelling in a few files where that changes legal wording which I am leery of us doing.

**Comment by mdavidsaver** [comment]
> Wow! Have you been having fun with this?

Fun with proofreading?  Not so much.  More like wanting to do this once and be done with it for awhile (like a decade or so).  I did have some fun with regexp to [automate](https://github.com/mdavidsaver/cspell) the process.

> I'm pretty sure Jeff still beat you in the total number of mis-spellings though.

Ah, well.  He didn't have Andrew Johnson looking over his shoulder.

>  found a few small issues and dictionary holes

fyi. I'm checking against the en_US dictionary of HunSpell 2018.04.16-1 (Debian package).

No objections to the un-corrections.  I'll hopefully get to this soon.

**Comment by mdavidsaver** [comment]
Un-corrections applied.

File: modules/ca/src/client/tcpiiu.cpp [hunk 13]
```diff
@@ -2060,7 +2060,7 @@ bool tcpiiu::bytesArePendingInOS () const
     }
     return false;
 #else
-    osiSockIoctl_t bytesPending = 0; /* shut up purifys yapping */
+    osiSockIoctl_t bytesPending = 0; /* shut up purifies yapping */
```
Comment (anjohnson) [inline modules/ca/src/client/tcpiiu.cpp:2063]: "Purify" was a commercial lint-like tool, so this should be `purify's`

File: modules/libcom/src/misc/shareLib.h [hunk 14]
```diff
@@ -74,7 +74,7 @@
  *
  * By default shareLib.h sets the DLL import / export keywords to import from
  * a DLL so that, for DLL consumers (users), nothing special must be done. However,
- * DLL implementors must set epicsExportSharedSymbols as above to specify
+ * DLL implements must set epicsExportSharedSymbols as above to specify
```
Comment (anjohnson) [inline modules/libcom/src/misc/shareLib.h:77]: Let's do `implementers` instead here.

**Review by anjohnson — APPROVED** [review]
A couple more here, but otherwise this looks good.

**Comment by mdavidsaver** [comment]
Applied as e34b6c5c0ce5a112fe79bc34b01401ce64fed9c4

**Pull request closed without merging**